pinot-core/src/main/java/org/apache/pinot/core/operator/BaseOperator.java

```diff
@@ -52,4 +54,25 @@ public final T nextBlock() {
 
   // Make it protected because we should always call nextBlock()
   protected abstract T getNextBlock();
+
+  // Enforcing sub-class to implement the getOperatorName(), as they can just return a static final,
+  // as opposed to this super class calling getClass().getSimpleName().
+  public abstract String getOperatorName();
+
+  public abstract String getExplainPlanName();
+
+  @Override
+  public List<Operator> getChildOperators() {
+    return new ArrayList<>();
```

Comment:
Collections.emptyList() might be better ?

Reply:
I also feel that not making this abstract has the potential for problems if someone who is writing a new operator misses out on implementing this interface (because compiler won't complain) but the operator has one or more child operators. So I would suggest to make this abstract and force every operator to implement this even if it has to return empty list

Reply:
Fixed. `getChildOperators()` is abstract now.